packages/graph/hash_graph/lib/graph/src/store/query/mod.rs

```diff
@@ -14,5 +16,45 @@ pub use self::{
 /// [`store`]: crate::store
 // TODO: Implement for `DataType`, `PropertyType`, etc. when `Path` is implemented
 pub trait QueryRecord {
-    type Path<'q>: TryFrom<Path>;
+    type Path<'q>: TryFrom<Path> + RecordPath;
+}
+
+pub trait RecordPath {
```

Comment:
I had something like `TypedPath` in mind, but this would imply something like `UntypedPath` (or `Path`), which in terms requires specialization. As I expect us to do further validation at some later point I think a name with `Path` in its name makes more sense, but I also don't mind changing it.

Reply:
We can revisit if you think this is going to change